### ISSUE TITLE
[PyTorch Upstream]  TIMM model "pnasnet5large" E2E training  bfloat16 accuracy regression.

### ISSUE THREAD

**Issue by etaf** [intel/intel-xpu-backend-for-triton#1353]
After PyTorch v2.4 branch cut, we continue to enable new benchmarks of TIMM and Torchbench on PyTorch master. When enabling TIMM new test scope by upgrade triton commit pin from
[b8c64f6](https://github.com/intel/intel-xpu-backend-for-triton/commit/b8c64f64c18d8cac598b3adb355c21e7439c21de)
to [aac14a3](https://github.com/intel/intel-xpu-backend-for-triton/commit/aac14a3b93f11d781d1d5ebc5400b15ae8df5185)
， we found a TIMM model pnasnet5large accuracy regression on dtype bfloat16.

#### Driver：
LTS 803.61

#### Stock pytorch:
- main branch
- Apply this PR to https://github.com/pytorch/pytorch/pull/126516 Pytorch （ because onednn doesn't support fp64 GEMM，this PR makes the accuracy check align with CUDA）

#### TIMM env
```
TIMM_COMMIT_ID=730b907b4d45a4713cbc425cbf224c46089fd514                    #pytorch/.ci/docker/ci_commit_pins/timm.txt
TORCHVISION_COMMIT_ID=d23a6e1664d20707c11781299611436e1f0c104f      #pytorch/.github/ci_commit_pins/vision.txt
git clone --single-branch -b main https://github.com/pytorch/vision.git
cd vision && git checkout $TORCHVISION_COMMIT_ID
python setup.py bdist_wheel && pip uninstall torchvision -y && pip install dist/*.whl
# install timm without dependencies
pip install --no-deps git+https://github.com/huggingface/pytorch-image-models@$TIMM_COMMIT_ID
# install timm dependencies without torch and torchvision
pip install $(curl -sSL https://raw.githubusercontent.com/huggingface/pytorch-image-models/$TIMM_COMMIT_ID/requirements.txt | grep -vE torch)
```

#### Triton version:
b8c64f64c18d8cac598b3adb355c21e7439c21de: pass
aac14a3b93f11d781d1d5ebc5400b15ae8df5185: fail_accuracy
#### command:
Test script: https://github.com/intel/torch-xpu-ops/blob/main/.github/scripts/inductor_xpu_test.sh
```
bash bash inductor_xpu_test.sh timm_models bfloat16 training accuracy xpu 0 static 1 0 pnasnet5large
```
